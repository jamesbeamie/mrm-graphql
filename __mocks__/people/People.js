const allPeopleReturnData = {
  data: {
    users: {
      users: [
        {
          email: 'michael.mukalo@andela.com',
          name: 'Michael Mukalo',
          picture: 'https://lh4.googleusercontent.com/-nvDSOOB7tuI/AAAAAAAAAAI/AAAAAAAAABA/otjsk3lztyc/photo.jpg?sz=50',
          location: 'Nairobi',
          id: '5',
          roles: [
            {
              role: 'Admin',
              id: '2',
              __typename: 'Role',
            },
          ],
          __typename: 'User',
        },
        {
          email: 'marlone.akidiva@andela.com',
          name: 'Marlone Akidiva',
          picture: 'https://lh4.googleusercontent.com/-2TgU0EHynu8/AAAAAAAAAAI/AAAAAAAAAAc/ML95I5e2k_c/photo.jpg?sz=50',
          location: 'Nairobi',
          id: '6',
          roles: [
            {
              role: 'Admin',
              id: '2',
              __typename: 'Role',
            },
          ],
          __typename: 'User',
        },
        {
          email: 'george.tibetegya@andela.com',
          name: 'George Tibetegya',
          picture: 'https://lh3.googleusercontent.com/-yhiUccQX5FE/AAAAAAAAAAI/AAAAAAAAAAc/fsBnDfr2fu8/photo.jpg?sz=50',
          location: 'Kampala',
          id: '7',
          roles: [
            {
              role: 'Admin',
              id: '2',
              __typename: 'Role',
            },
          ],
          __typename: 'User',
        },
      ],
      __typename: 'Users',
      pages: 2,
      queryTotal: 11,
      hasNext: true,
      hasPrevious: false,
    },
  },
};

export { allPeopleReturnData as default };
