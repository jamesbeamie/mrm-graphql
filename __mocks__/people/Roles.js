const allRolesReturnData = {
  data: {
    roles: [
      {
        id: '1',
        role: 'Default User',
      },
      {
        id: '2',
        role: 'Admin',
      },
    ],
  },
};

export { allRolesReturnData as default };
