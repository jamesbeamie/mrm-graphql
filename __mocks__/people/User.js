const allUserReturnData = {
  data: {
    user: {
      id: 8,
      location: 'Lagos',
    },
  },
};

export { allUserReturnData as default };

