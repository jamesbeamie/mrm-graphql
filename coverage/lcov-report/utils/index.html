<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for utils</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../index.html">All files</a> utils
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">93.27% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>97/104</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">80.95% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>17/21</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">88% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>22/25</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">94.9% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>93/98</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="Constants.js"><a href="Constants.js.html">Constants.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	</tr>

<tr>
	<td class="file high" data-value="Cookie.js"><a href="Cookie.js.html">Cookie.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="25" class="abs high">25/25</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="10" class="abs high">10/10</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="23" class="abs high">23/23</td>
	</tr>

<tr>
	<td class="file high" data-value="Utilities.js"><a href="Utilities.js.html">Utilities.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="40" class="abs high">40/40</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="8" class="abs high">8/8</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="40" class="abs high">40/40</td>
	</tr>

<tr>
	<td class="file high" data-value="activityData.js"><a href="activityData.js.html">activityData.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	</tr>

<tr>
	<td class="file high" data-value="checkDate.js"><a href="checkDate.js.html">checkDate.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	</tr>

<tr>
	<td class="file medium" data-value="formatSetupData.js"><a href="formatSetupData.js.html">formatSetupData.js</a></td>
	<td data-value="72" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 72%;"></div><div class="cover-empty" style="width:28%;"></div></div></td>
	<td data-value="72" class="pct medium">72%</td>
	<td data-value="25" class="abs medium">18/25</td>
	<td data-value="55.56" class="pct medium">55.56%</td>
	<td data-value="9" class="abs medium">5/9</td>
	<td data-value="62.5" class="pct medium">62.5%</td>
	<td data-value="8" class="abs medium">5/8</td>
	<td data-value="78.26" class="pct medium">78.26%</td>
	<td data-value="23" class="abs medium">18/23</td>
	</tr>

<tr>
	<td class="file high" data-value="history.js"><a href="history.js.html">history.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	</tr>

<tr>
	<td class="file high" data-value="notification.js"><a href="notification.js.html">notification.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	</tr>

<tr>
	<td class="file high" data-value="roomSetupMock.js"><a href="roomSetupMock.js.html">roomSetupMock.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	</tr>

<tr>
	<td class="file high" data-value="routes.js"><a href="routes.js.html">routes.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu May 02 2019 21:44:28 GMT+0300 (East Africa Time)
</div>
</div>
<script src="../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../sorter.js"></script>
</body>
</html>
