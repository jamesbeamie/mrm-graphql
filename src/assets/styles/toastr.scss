/* Note that this is toastr v2.1.3, the "latest" version in url has no more maintenance,
* please go to https://cdnjs.com/libraries/toastr.js and pick a certain version you want to use,
* make sure you copy the url from the website since the url may change between versions.
* */
@import './variables.scss';

.toast-message {
  -ms-word-wrap: break-word;
  word-wrap: break-word;
  font-family: "PT Sans Caption", sans-serif;
 }
 .toast-message a,
 .toast-message label {
  color: map-get($color-white, 000);
 }
 .toast-close-button {
  position: relative;
  right: -0.3em;
  top: -0.7em;
  float: right;
  font-size: 20px;
  color: map-get($color-black, 900);
  opacity: 0.8;
  -ms-filter: progid:DXImageTransform.Microsoft.Alpha(Opacity=80);
  filter: alpha(opacity=80);
  line-height: 1;
 }
 .rtl .toast-close-button {
  left: -0.1em;
  float: left;
  right: 0.3em;
 }
 button.toast-close-button {
  padding: 0;
  cursor: pointer;
  background: 0 0;
  border: 0;
  -webkit-appearance: none;
 }
 .toast-top-center {
  top: 0;
  right: 0;
  width: 100%;
 }
 #toast-container {
  position: fixed;
  z-index: 999999;
  pointer-events: none;
 }
 #toast-container * {
  -moz-box-sizing: border-box;
  -webkit-box-sizing: border-box;
  box-sizing: border-box;
 }
 #toast-container > div {
  position: relative;
  pointer-events: auto;
  margin: 20px 0 6px;
  padding: 21px 21px 20px 20px;
  -moz-border-radius: 3px;
  -webkit-border-radius: 3px;
  border-radius: 3px;
  background-position: 15px center;
  background-repeat: no-repeat;
  -moz-box-shadow: 0 0 12px map-get($color-black, 100);
  -webkit-box-shadow: 0 0 12px map-get($color-black, 100);
  box-shadow: 0 0 12px map-get($color-black, 100);
  -ms-filter: progid:DXImageTransform.Microsoft.Alpha(Opacity=80);
  filter: alpha(opacity=80);
 }
 #toast-container > div.rtl {
  direction: rtl;
  padding: 15px 50px 15px 15px;
  background-position: right 15px center;
 }
 #toast-container.toast-top-center > div {
  width: 400px;
  margin-left: auto;
  margin-right: auto;
 }
 .toast {
  background-color: map-get($color-black, 900);
 }
 .toast-success {
  background-color: $color-success;
 }
 .toast-error {
  background-color: $color-danger;
  color: $toast-error;
 }
 @media all and (max-width: 240px) {
  #toast-container > div {
    padding: 8px 8px 8px 50px;
    width: 11em;
  }
  #toast-container > div.rtl {
    padding: 8px 50px 8px 8px;
  }
  #toast-container .toast-close-button {
    right: -0.2em;
    top: -0.2em;
  }
  #toast-container .rtl .toast-close-button {
    left: -0.2em;
    right: 0.2em;
  }
 }
 @media all and (min-width: 241px) and (max-width: 480px) {
  #toast-container > div {
    padding: 8px 8px 8px 50px;
    width: 18em;
  }
  #toast-container > div.rtl {
    padding: 8px 50px 8px 8px;
  }
  #toast-container .toast-close-button {
    right: -0.2em;
    top: -0.2em;
  }
  #toast-container .rtl .toast-close-button {
    left: -0.2em;
    right: 0.2em;
  }
 }
 @media all and (min-width: 481px) and (max-width: 768px) {
  #toast-container > div {
    padding: 15px 15px 15px 50px;
    width: 25em;
  }
  #toast-container > div.rtl {
    padding: 15px 50px 15px 15px;
  }
 }