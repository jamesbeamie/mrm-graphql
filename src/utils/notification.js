/* eslint-disable */
/**
 * toastr => This is the toastr module imported from 'toastr' npm package
 * notifyType => This is the notification type (success or error)
 * message => This is the notification message
 *
 * How to use this function
 * const notify = notification(toastr, 'success', 'noficationMessage');
 * notify()
 */
const notification = (toastr, notifyType, message) => {
  toastr['options'] = {
    'closeButton': true,
    'positionClass': 'toast-top-center',
    'preventDuplicates': true,
    'onclick': null,
    'showDuration': '300',
    'hideDuration': '1000',
    'timeOut': '2000',
    'extendedTimeOut': '100',
    'showEasing': 'swing',
    'hideEasing': 'linear',
    'showMethod': 'fadeIn',
    'hideMethod': 'fadeOut',
    "newestOnTop": true,
  };
  return () => toastr[notifyType](message);
};

export default notification;
