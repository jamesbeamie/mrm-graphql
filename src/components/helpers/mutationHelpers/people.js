/* eslint-disable import/prefer-default-export */
import apolloClient from '../../../utils/ApolloClient';
import { INVITE_PERSON_MUTATION, CHANGE_USER_LOCATION_MUTATION } from '../../../graphql/mutations/People';
import { decodeTokenAndGetUserData } from '../../../utils/Cookie';
import { GET_USER_QUERY } from '../../../graphql/queries/People';
import GET_ALL_LEVELS from '../../../graphql/queries/Levels';
import { GET_DEVICES_QUERY } from '../../../graphql/queries/Devices';


const invitePersonMutation = async (email, client = apolloClient) => {
  await client
    .mutate({
      mutation: INVITE_PERSON_MUTATION,
      name: 'invitePersonMutation',
      variables: { email },
    });
};

const changeUserLocation = async (locationId, queryEmail, client = apolloClient) => {
  const { UserInfo: { email: tokenEmail } } = decodeTokenAndGetUserData();
  const email = queryEmail || tokenEmail;

  const { data } = await client
    .mutate({
      mutation: CHANGE_USER_LOCATION_MUTATION,
      name: 'changeUserLocation',
      variables: { email, locationId },
      refetchQueries: [
        { query: GET_USER_QUERY, variables: { email } },
        { query: GET_ALL_LEVELS },
        { query: GET_DEVICES_QUERY },
      ],
    });

  return data;
};

export { invitePersonMutation, changeUserLocation };
