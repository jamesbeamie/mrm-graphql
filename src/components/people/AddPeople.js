import React from 'react';
import PropTypes from 'prop-types';
import toastr from 'toastr';
import ActionButtons from '../commons/ActionButtons';
import MrmModal from '../commons/Modal';
import { Input } from '../commons';
import { invitePersonMutation } from '../helpers/mutationHelpers/people';
import notification from '../../utils/notification';

/**
 * Add Resource Component
 *
 * @extends React.Component
 *
 * @returns {JSX}
 */
class AddPeople extends React.Component {
  state = {
    people: '',
    closeModal: false,
  };

  /**
   * Ensures that the modal for inviting a person closes
   * when a user hits CANCEL on the modal or when
   * inviting a person is successful
   *
   * @returns {void}
   */
  handleCloseModal = () => {
    this.setState({ closeModal: true });
  };

  /**
   * Ensures that the state is updated basing on the
   * user input
   *
   * @param {object} event The input event parameter
   * @param {number} capacity
   *
   * @returns {void}
   */
  handleInputChange = (event) => {
    const { name, value } = event.target;
    this.setState({ people: value, [name]: value });
  };

  validateUser = (people) => {
    const { users } = this.props.availableUsers;
    const existingUsers = users || [];
    return existingUsers.some(item => item.email === people);
  }

  submitInvite = () => {
    const { email } = this.state;
    if (this.validateUser(email)) {
      notification(toastr, 'error', 'Provide a new, unique user to invite')();
    } else {
      invitePersonMutation(email)
        .then(() => {
          notification(toastr, 'success', 'User Successfully invited')();
        })
        .catch((error) => {
          notification(toastr, 'error', error.graphQLErrors[0].message)();
        });
    }
  };

  /**
   * It updates the state value of closeModal
   * to false whenever the modal closes
   *
   * @returns {void}
   */
  handleModalStateChange = () => {
    this.state.closeModal && this.setState({ closeModal: false });
  };

  render() {
    const {
      people,
      closeModal,
    } = this.state;
    return (
      <MrmModal
        title="INVITE A NEW USER"
        buttonText="Invite a New User"
        closeModal={closeModal}
        handleCloseRequest={this.handleModalStateChange}
        className="add-resource-modal"
      >
        <form className="amenity-form" onSubmit={this.handleCloseModal}>
          <div>
            <Input
              id="user"
              name="email"
              placeholder="Enter Users Email"
              labelName="Users Email Address"
              labelClass="add-resource-controls"
              value={people}
              onChange={this.handleInputChange}
            />
            <div className="loading-btn-div">
              <ActionButtons
                withCancel
                onClickCancel={this.handleCloseModal}
                actionButtonText="SEND INVITE"
                onClickSubmit={this.submitInvite}
              />
            </div>
          </div>
        </form>
      </MrmModal>
    );
  }
}

AddPeople.propTypes = {
  availableUsers: PropTypes.oneOfType([PropTypes.object, PropTypes.array]).isRequired,
};

export default AddPeople;
