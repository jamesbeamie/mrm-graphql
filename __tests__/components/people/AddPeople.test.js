import React from 'react';
import { shallow } from 'enzyme';
import toastr from 'toastr';
import { ApolloError } from 'apollo-client';
import wait from 'waait';
import * as notification from '../../../src/utils/notification';
import AddPeople from '../../../src/components/people/AddPeople';
import { invitePersonMutation } from '../../../src/components/helpers/mutationHelpers/people';

jest.mock('../../../src/components/helpers/mutationHelpers/people');

describe('AddPeople Component', () => {
  const initProps = {
    availableUsers: {},
  };

  const errorMessage = 'User already joined Converge';
  const error = new ApolloError({ graphQLErrors: [new Error(errorMessage)] });
  const notificationSpy = jest.spyOn(notification, 'default');
  const wrapper = shallow(<AddPeople {...initProps} />);

  it('renders properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should change closeModal to true when handleCloseModal is called', () => {
    wrapper.setState({ closeModal: false });
    wrapper.instance().handleCloseModal();
    expect(wrapper.state('closeModal')).toEqual(true);
  });

  it('should call  handleCloseModal()', () => {
    const modalSpy = jest.spyOn(wrapper.instance(), 'handleCloseModal').mockImplementation(() => {});
    wrapper.instance().handleCloseModal();
    expect(modalSpy).toHaveBeenCalled();
  });

  it('should handleModalStateChange()', () => {
    wrapper.instance().handleModalStateChange();
    expect(wrapper.state('closeModal')).toBe(false);
  });

  it('should call notification with a success message', async () => {
    const instance = wrapper.instance();
    invitePersonMutation.mockResolvedValue({});
    instance.submitInvite();
    await wait();
    expect(notificationSpy).toHaveBeenCalled();
    expect(notificationSpy).toHaveBeenCalledWith(toastr, 'success', 'User Successfully invited');
  });

  it('should call notification with an error message', async () => {
    const instance = wrapper.instance();
    invitePersonMutation.mockRejectedValue(error);
    instance.submitInvite();
    await wait();
    expect(notificationSpy).toHaveBeenCalled();
    expect(notificationSpy).toHaveBeenCalledWith(toastr, 'error', errorMessage);
  });

  it('should change the value of people in the state', () => {
    expect(wrapper.instance().state.people).toEqual('');
    const event = { target: { name: 'email', value: 'johndoe@andela.com' } };
    wrapper.instance().handleInputChange(event);
    expect(wrapper.instance().state.people).toEqual('johndoe@andela.com');
  });
});
