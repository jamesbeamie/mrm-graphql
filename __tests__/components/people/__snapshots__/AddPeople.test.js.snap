// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddPeople Component renders properly 1`] = `
<MrmModal
  buttonText="Invite a New User"
  className="add-resource-modal"
  closeModal={false}
  handleCloseRequest={[Function]}
  modalButtonClassName="button"
  openModal={false}
  title="INVITE A NEW USER"
>
  <form
    className="amenity-form"
    onSubmit={[Function]}
  >
    <div>
      <Input
        controlsClass=""
        id="user"
        inputClass="mrm-input default-input"
        labelClass="add-resource-controls"
        labelName="Users Email Address"
        name="email"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Enter Users Email"
        type="text"
        value=""
      />
      <div
        className="loading-btn-div"
      >
        <ActionButtons
          actionButtonText="SEND INVITE"
          cancelButtonText="CANCEL"
          isLoading={false}
          onClickCancel={[Function]}
          onClickSubmit={[Function]}
          showActionButton={true}
          withCancel={true}
        />
      </div>
    </div>
  </form>
</MrmModal>
`;
